task3/task3_solution.py

```diff
@@ -0,0 +1,124 @@
+class Journal:
+
+    def __init__(self, students):
+        self.student_data = students
+
+    def to_letters(self, grades):
+        res = ""
+        for grade in grades:
+            if grade >= 95.0:
+                res += "A"
+            elif grade >= 85.0:
+                res += "B"
+            elif grade >= 75.0:
+                res += "C"
+            elif grade >= 65.0:
+                res += "D"
+            elif grade >= 60.0:
+                res += "E"
+            elif grade < 60.0:
+                res += "F"
+            res += " "
+        return res.strip()
+
+    # calc overall average grade
+    # not universal
+    def avg_grade(self, homework, quizzes, tests):
+        grades = homework + quizzes + tests
+        res = sum(grades)/len(grades)
+        return f"{res:.2f}"
+
+    def get_letter_grades(self, *args, throw_exception=False):
+        for dict in self.student_data:
+            if dict.get("name", 0) in args:
+                print("Grades of " + dict.get("name", 0) + ": ")
+                print("Homework:", self.to_letters(dict.get("homework", 0)))
+                print("Quizzes:", self.to_letters(dict.get("quizzes", 0)))
+                print("Tests:", self.to_letters(dict.get("tests", 0)))
+            else:
+                # if keyword specified, show error instead of skipping
+                if throw_exception:
+                    print("No student with such name in table")
+
+    def get_student_average(self, *args, throw_exception=False):
+        for dict in self.student_data:
+            if dict.get("name", 0) in args:

```

Comment:
0 doesn't seem to fit here, you are looking for strings and getting 0 to search for
You can either leave it as default None, or change it to string

Reply:
This if is meant to evaluate to 0 if name is not found, therefore executing else statement with throw_exception, and that's where I'm experiencing troubles.

Reply:
Yeah, but you can just leave `None` or `''` and you'll get same result but it would make much more sense then `0` in that case